packages/block-library/src/gallery/edit.js

```diff
@@ -110,6 +120,20 @@ function GalleryEdit( props ) {
 		onFocus,
 	} = props;
 
+	const [ blockSetting ] = useSettings( 'blocks' );
+
+	const lightboxSetting = blockSetting.hasOwnProperty( 'core/image' )
+		? blockSetting[ 'core/image' ]?.lightbox
+		: false;
+
+	useEffect( () => {
+		if ( ! lightboxSetting?.allowEditing ) {
+			linkOptions = linkOptions.filter(
+				( option ) => option.value !== LINK_DESTINATION_LIGHTBOX
+			);
+		}
+	}, [ lightboxSetting?.allowEditing ] );
```

Comment:
I think unnecessary `useEffect` should be avoided. Additionally, I think we need to check `enabled` as well, not just `allowEditing`.

```javascript
	const showLightboxOption =
		lightboxSetting?.enabled && lightboxSetting?.allowEditing;
	const filteredLinkOptions = showLightboxOption
		? linkOptions
		: linkOptions.filter(
				( option ) => option.value !== LINK_DESTINATION_LIGHTBOX
		  );
```

In other words, I'd expect the lightbox to be disabled when the following theme.json is defined:

```json
{
	"version": 3,
	"settings": {
		"blocks": {
			"core/image": {
				"lightbox": {
					"enabled": false
				}
			}
		}
	}
}
```

However, in the Image block, the lightbox does not seem to be disabled even if the above theme.json is defined. This seems like a bug to me.

Reply:
That's right, I was mistaken 😅